run-tests.php

```diff
@@ -65,122 +87,673 @@
+
+    --verbose
+    -v          Verbose mode.
+
+    --help
+    -h          This Help.
+
+    --html <file> Generate HTML output.
+
+    --temp-source <sdir>  --temp-target <tdir> [--temp-urlbase <url>]
+                Write temporary files to <tdir> by replacing <sdir> from the
+                filenames to generate with <tdir>. If --html is being used and
+                <url> given then the generated links are relative and prefixed
+                with the given url. In general you want to make <sdir> the path
+                to your source files and <tdir> some pach in your web page
+                hierarchy with <url> pointing to <tdir>.
+
+    --keep-[all|php|skip|clean]
+                Do not delete 'all' files, 'php' test file, 'skip' or 'clean'
+                file.
+
+    --set-timeout [n]
+                Set timeout for individual tests, where [n] is the number of
+                seconds. The default value is 60 seconds, or 300 seconds when
+                testing for memory leaks.
+
+    --show-[all|php|skip|clean|exp|diff|out]
+                Show 'all' files, 'php' test file, 'skip' or 'clean' file. You
+                can also use this to show the output 'out', the expected result
+                'exp' or the difference between them 'diff'. The result types
+                get written independent of the log format, however 'diff' only
+                exists when a test fails.
+
+    --show-slow [n]
+                Show all tests that took longer than [n] milliseconds to run.
+
+    --no-clean  Do not execute clean section if any.
+
+HELP;
+                        exit(1);
+                }
+            }
+
+            if (!$is_switch) {
+                $testfile = realpath($argv[$i]);
+
+                if (!$testfile && strpos($argv[$i], '*') !== false && function_exists('glob')) {
+
+                    if (preg_match("/\.phpt$/", $argv[$i])) {
+                        $pattern_match = glob($argv[$i]);
+                    } else if (preg_match("/\*$/", $argv[$i])) {
+                        $pattern_match = glob($argv[$i] . '.phpt');
+                    } else {
+                        die("bogus test name " . $argv[$i] . "\n");
+                    }
+
+                    if (is_array($pattern_match)) {
+                        $test_files = array_merge($test_files, $pattern_match);
+                    }
+
+                } else if (is_dir($testfile)) {
+                    find_files($testfile);
+                } else if (preg_match("/\.phpt$/", $testfile)) {
+                    $test_files[] = $testfile;
+                } else {
+                    die("bogus test name " . $argv[$i] . "\n");
+                }
+            }
+        }
+
+        if (strlen($conf_passed)) {
+            if (substr(PHP_OS, 0, 3) == "WIN") {
+                $pass_options .= " -c " . escapeshellarg($conf_passed);
+            } else {
+                $pass_options .= " -c '$conf_passed'";
```

Comment:
Why no `escapeshellarg` here?

Reply:
I didn't write this code, so I don't know.